protobuf/lib/src/protobuf/generated_message.dart

```diff
@@ -310,6 +310,8 @@ abstract class GeneratedMessage {
   /// The tagNumber should be a valid tag or extension.
   void clearField(int tagNumber) => _fieldSet._clearField(tagNumber);
 
+  /// For generated code only.
+  /// @nodoc
   int $_whichOneof(int oneofIndex) => _fieldSet._oneofCases![oneofIndex] ?? 0;
```

Comment:
Consider also marking these as @protected. They will give a warning if used in the wrong context, and if https://github.com/dart-lang/sdk/issues/49273 was fixed they would also be removed from tab-completion.

Reply:
I need to make a pass to find out which ones need `@Protected`, I'd like to do that in a separate PR if that's OK.

Reply:
That's fine!